src/providers/eth_provider/database/ethereum.rs

```diff
@@ -56,6 +64,17 @@ impl EthereumTransactionStore for Database {
         let filter = EthDatabaseFilterBuilder::<filter::Transaction>::default().with_tx_hash(&transaction.hash).build();
         Ok(self.update_one(StoredTransaction::from(transaction), filter, true).await?)
     }
+
+    #[instrument(skip_all, name = "db::upsert_transaction_receipt", err)]
+    async fn upsert_transaction_receipt(
+        &self,
+        transaction: WithOtherFields<TransactionReceipt>,
+    ) -> Result<(), EthApiError> {
+        let filter =
+            EthDatabaseFilterBuilder::<filter::Receipt>::default().with_tx_hash(&transaction.transaction_hash).build();
+        let stored_receipt = StoredTransactionReceipt { receipt: transaction };
+        Ok(self.update_one(stored_receipt, filter, true).await?)
+    }
```

Comment:
I'd rather not add methods here if we only use them for tests.

Reply:
agree, moved to katana. thanks